components/welcome-mat/index.jsx

```diff
@@ -0,0 +1,243 @@
+	 * * `title`: Title for the Welcome Mat
+	 * * `description`: Label for the radio input
+	 */
+	labels: PropTypes.shape({
+		title: PropTypes.string,
+		description: PropTypes.string,
+	}),
+	/**
+	 *	Variant of the WelcomeMat
+	 */
+	variant: PropTypes.oneOf([
+		'steps',
+		'info-only',
+		'splash',
+		'trailhead-connected',
+	]),
+	/**
+	 * Link to learn more button
+	 */
+	onRenderInfoActions: PropTypes.func,
+	/**
+	 *  Infobadge for trailhead variant
+	 */
+	infoBadge: PropTypes.func,
+	/**
+	 *  Do not show again checkbox for info-only variant
+	 */
+	doNotShowAgainCheckbox: PropTypes.func,
+};
+
+const defaultProps = {
+	variant: 'steps',
+	isOpen: true,
+};
+
+/**
+ * A Welcome Mat provides a series of unordered items a user can click to learn about a thematic topic.
+ */
+class WelcomeMat extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			completedSteps: 0,
+			totalSteps: 0,
+			progress: 0,
+		};
+	}
+
+	componentWillMount() {
+		this.generatedId = shortid.generate();
+		this.getCount();
+	}
+
+	/**
+	 * Get the WelcomeMat's HTML id. Generate a new one if no ID present.
+	 */
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	getCount() {
+		const totalSteps = React.Children.count(this.props.children);
+		const completedSteps = React.Children.toArray(this.props.children).filter(
+			(c) => c.props.isComplete
+		).length;
+		const progress = completedSteps / totalSteps * 100;
+		this.setState({
+			totalSteps,
+			completedSteps,
+			progress,
+		});
+	}
+
+	render() {
+		const splash = (
+			<div
+				className={classNames(
+					'slds-welcome-mat__info-content',
+					this.props.className
+				)}
+				id={`${this.getId()}-content`}
+			>
+				<h2
+					className="slds-welcome-mat__info-title"
+					id={`${this.getId()}-label`}
+				>
+					{this.props.labels.title}
+				</h2>
+				<div className="slds-welcome-mat__info-description slds-text-longform">
+					<p>{this.props.labels.description}</p>
+				</div>
+				{this.props.variant === 'info-only' ||
+				this.props.variant === 'splash' ? (
+					<div className="slds-welcome-mat__info-actions">
+						{this.props.onRenderInfoActions()
+							? this.props.onRenderInfoActions()
+							: null}
+						<div className="slds-m-top_large">
+							{this.props.doNotShowAgainCheckbox()
+								? this.props.doNotShowAgainCheckbox()
+								: null}
+						</div>
+					</div>
+				) : null}
+				{(this.props.variant === 'steps' ||
+					this.props.variant === 'trailhead-connected') &&
+				this.props.children ? (
+					<React.Fragment>
+						<div
+							className={classNames(
+								'slds-welcome-mat__info-progress',
+								// eslint-disable-next-line eqeqeq
+								this.state.completedSteps == this.state.totalSteps
+									? 'slds-welcome-mat__info-progress_complete'
+									: null
+							)}
+						>
+							{this.props.variant === 'trailhead-connected'
+								? React.Children.map(this.props.infoBadge(), (child) =>
```

Comment:
Since `props.infoBadge` will be changed to `PropTypes.node`, this function rendering and mapping will no longer be necessary. Instead, continue with `React.cloneElement(this.props.infoBadge, { /* ...etc */` from the next line